### PR TITLE
Merge main into Api adoption

### PR DESCRIPTION
**Pull request by mortalisk**

### CODE REVIEW

**Comment by oysteoh** [comment]
This was a quite big PR! :sweat_smile: I see most of it at least does not relate to files where we have done anything so my assumptions is that most of it is ok. Rest of it seems ok from my point of view, cannot see anything special that is not correct. But we should also have some more eyes on it before merging. @DanSava ?

File: ert_gui/gert_main.py [hunk 1]
```diff
@@ -62,20 +64,18 @@ def run_gui(args):
...
     args.config = os.path.basename(args.config)
     ert = EnKFMain(res_config, strict=True, verbose=args.verbose)
-    with Storage.connect_or_start_server(
-        res_config=os.path.basename(args.config)
-    ) as storage:
+    with Storage.connect_or_start_server(res_config=os.path.basename(args.config)):
         # window reference must be kept until app.exec returns
         notifier = ErtNotifier(args.config)
-        window = _start_window(ert, notifier, args)
+        _start_window(ert, notifier, args)
```
Comment (DanSava) [inline ert_gui/gert_main.py:70]: not sure if this is OK given the comment `# window reference must be kept until app.exec returns` I am sure it is fine but if it is not then we will get some very strange errors that are hard to debug, so I would keep 
```
window = _start_window(ert, notifier, args) # noqa
```

Comment (berland) [reply ert_gui/gert_main.py:70]: This is a mixup with the flake8 commit, where this change has been reverted. Should go away when rebased on main.

Comment (berland) [reply ert_gui/gert_main.py:70]: https://github.com/equinor/ert/commit/b06303d5f2011495da1e70f7da972cc4ad0e1998

**Review by DanSava — COMMENTED** [review]
Had one small comment but otherwise, it looks good!

File: ert_gui/gert_main.py (hunk 1 above)
Comment (berland) [reply ert_gui/gert_main.py:70]: Not needed at all, remove if you wish.

Comment (DanSava) [reply ert_gui/gert_main.py]: ok, I thought it was some obscure new flake8 thing :smile:

**Review by DanSava — APPROVED** [review]
:+1:  looks good!

**Pull request merged**